src/Symfony/Component/Console/Command/LockableTrait.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Console\Command;
+
+use Symfony\Component\Console\Exception\LogicException;
+use Symfony\Component\Console\Exception\RuntimeException;
+use Symfony\Component\Filesystem\LockHandler;
+
+/**
+ * Basic lock feature for commands.
+ *
+ * @author Fabien Potencier <fabien@symfony.com>
+ */
+trait LockableTrait
+{
+    protected $lockHandler;
+
+    /**
+     * Locks a command.
+     *
+     * @return bool
+     */
+    protected function lock($name = null, $blocking = false)
```

Comment:
Well, every class using this trait will now expose the used methods as an extension point which might not be what the user intended.